### ISSUE TITLE
Export DLP rules from Security and Compliance Center 

### ISSUE THREAD

**Issue by ACK00** [microsoft/Microsoft365DSC#1767]
Need a solution for exporting and importing Security and Compliance Center DLP policies. The provided cmdlets from this document: https://docs.microsoft.com/en-us/powershell/module/exchange/export-dlppolicycollection?view=exchange-ps
only partially work. The Export DlpCollection did not work. 

Any ideas? Have ran into this issue several times.

**Comment by NikCharlebois** [comment]
Have you tried exporting the DLP Policies from Microsoft365DSC? Are you running into any issues?

Export-M365DSCConfiguration -Components SCDLPCOmpliancePolicy

**Comment by ACK00** [comment]
> Have you tried exporting the DLP Policies from Microsoft365DSC? Are you running into any issues?
> 
> Export-M365DSCConfiguration -Components SCDLPCOmpliancePolicy

Am I running this from my admin PowerShell or do I need to be in the ExchangeOnline powershell? and I did just try to run this and no luck.

**Comment by andikrueger** [comment]
you need to run the command `Export-M365DSCConfiguration -Components SCDLPCOmpliancePolicy` in a standard PowerShell 5 session. This would be the complete script to export the policies:

```
# Generated by Microsoft365DSC from https://export.microsoft365dsc.com on 2/21/2022, 1:32:22 PM
# Visit https://microsoft365dsc.com for more information

# Getting client credential
$Credential = Get-Credential

# Exporting resources using credentials
Export-M365DSCConfiguration -Components @("SCDLPCompliancePolicy") -Credential $Credential
```

Additionally, please make sure to follow this guide: https://microsoft365dsc.com/user-guide/get-started/complete-story/

**Comment by ACK00** [comment]
@andikrueger thank you for the reply. Do I run this command in place of the Export-DlpPolicyCollection command. I am sorry but I have never used this resource before. Like I mentioned above, I Connect via IPPPSession to the security and compliance center with credentials and run the commands. In your instructions you hvae credential commands as well. and then the link to gettting started with 365dsc. So it looks like I have to install the dsc module to then follow the commands? Thank you for your assistance

**Comment by andikrueger** [comment]
Did you have a chance to look at the getting started guide for Microsoft 365 DSC? Microsoft365DSC handles the connection for you and there is no need to run the commands manually.

**Comment by ACK00** [comment]
@andikrueger yes I was able to set up the module and was able to connect, I rant the Export-m365DSCconfiguration command and it exported 1/3 rules. Now says Destination Path: and trying to set a path for the export, although I am not sure where the first export went without a file path.  See attached file. 

![image](https://user-images.githubusercontent.com/70402839/154993735-dbc83d26-e2d4-41da-9ab4-4b41d180cfbd.png)

**Comment by ACK00** [comment]
@andikrueger I was able to export 1/3 rules.... this is the Contoso demo site I am using.... now stuck on Destination path to see the configs... 

![image](https://user-images.githubusercontent.com/70402839/154994010-a9b64776-5f50-4c67-8007-024450fe7323.png)

**Comment by andikrueger** [comment]
No need to add set-content. Only the target path of the export is needed. Export-M365DSCConfiguration will handle the creation of the target folder and files

**Comment by ACK00** [comment]
@andikrueger Success! I have now gotten the policy configs in a file. Making progress, now my question is when I get to this command, how to I define what tenant I want the DLP policy configs to import to? 

Start-DSCConfiguration .\M365TenantConfig -Wait -Verbose -Force

**Comment by andikrueger** [comment]
You need to define some credentials to compile the configuration. These should be of the target tenant.

**Comment by ACK00** [comment]
@andikrueger define credentials in the .mof file?

**Comment by andikrueger** [comment]
No. They are used within the ps1 file and then compiled and therefor available within the mof file. If you need to secure the credentials in the mof file, there is a good docs article available: https://docs.microsoft.com/en-us/powershell/dsc/pull-server/securemof

**Comment by ACK00** [comment]
@andikrueger So do I literally change the credentials in the ps.1 file to the target tenant? For example, I export the policies into a mof file from tenant 1, I want to import them into tenant 2. So do I need to change the credentials in the mof file to tenant 2 and then run the Start-DSCConfiguration .\M365TenantConfig -Wait -Verbose -Force command?

**Comment by ACK00** [comment]
@andikrueger WE DID IT! Thank you so much for your help.

**Comment by ACK00** [comment]
@andikrueger  Got excited too soon, any custom DLP with custom sensitive info types we get errors. 

![image](https://user-images.githubusercontent.com/70402839/155182017-eaceaaf4-ccff-47a7-835b-c35eb571e88a.png)

**Comment by ACK00** [comment]
@desmay 

This is a rough template of replicated policies that I am tasked with importing. I understand this is not the ideal scenario for a DLP structure. 

![image](https://user-images.githubusercontent.com/70402839/155196963-858fee40-f19e-48c5-bd49-e66250e0ca3b.png)
![image](https://user-images.githubusercontent.com/70402839/155197029-e1c3bf8c-3537-4be4-93e8-2f1667b52359.png)
![image](https://user-images.githubusercontent.com/70402839/155197078-e00270ea-6703-44cc-96aa-895e1c8e1b87.png)
![image](https://user-images.githubusercontent.com/70402839/155197139-306e97f0-4b61-4359-adf1-8b240bec7cef.png)
![image](https://user-images.githubusercontent.com/70402839/155197231-86785385-8b6f-46cd-a759-3ac339b54979.png)

**Comment by desmay** [comment]
@ACK00 Thx let me see if I repro

**Comment by desmay** [comment]
@ACK00 After reviewing your screenshots we don't have support for custom SITs only the OOB SITs. We can look to add this to the backlog as an enhancement to SCDLPCompliancePolicy resource.

**Comment by gabriel-rech** [comment]
> you need to run the command `Export-M365DSCConfiguration -Components SCDLPCOmpliancePolicy` in a standard PowerShell 5 session. This would be the complete script to export the policies:
> 
> ```
> # Generated by Microsoft365DSC from https://export.microsoft365dsc.com on 2/21/2022, 1:32:22 PM
> # Visit https://microsoft365dsc.com for more information
> 
> # Getting client credential
> $Credential = Get-Credential
> 
> # Exporting resources using credentials
> Export-M365DSCConfiguration -Components @("SCDLPCompliancePolicy") -Credential $Credential
> ```
> 
> Additionally, please make sure to follow this guide: https://microsoft365dsc.com/user-guide/get-started/complete-story/

I just wanna say: if it works, I love you

**Comment by FabienTschanz** [comment]
Support for custom dlp sensitive information types should work.
